DESIGN.md

```diff
@@ -325,18 +325,149 @@ In the addition of the previous rules, an operator for `Flowable`:
+
+In concert, `create(OnSubscribe)` will not be available; standard operators extend the base types directly. The conversion of other RS-based libraries will happen through the `Flowable.wrap(Publisher<T>)` static method. 
+
+(*The unfortunate effect of `create` in 1.x was the ignorance of the Observable contract and beginner's first choice as an entry point. We can't eliminate this path since `rs.Publisher` is a single method functional interface that can be implemented just as badly.*)
 
-Flowable.create(AsyncGenerator generator)
+Therefore, new standard factory methods will try to address the common entry point requirements: 
+  - `create(SyncGenerator<T, S>)` to safe, synchronous generation of signals, one-by-one
+  - `create(AsyncGenerator<T, S>)` to batch-create signals based on request patterns
+  - `create(Consumer<? super SingleEmitter<T>>)` to relay a single value or error from other reactive sources (i.e., addListener callbacks)
+  - `create(Consumer<? super FlowEmitter<T>>)` to relay multiple values or error from multi-valued reactive-sources (i.e., button-clicks) while also give flow control options right there (buffer, drop, error, etc.).
+  - `create(Consumer<? super CompletionEmitter)>` signal a completion or error from valueless reactive sources
+
+The following table lists where these create methods will be available in respect of the base types:
+
+| Method | Flowable | Observable | Single | Completable |
+|--------|----------|------------|--------|-------------|
+| `create(SyncGenerator<T, S>)` | Yes | Yes | No | No |
+| `create(AsyncOnSubscribe<T, S>)` | Yes | No | No | No |
+| `create(Consumer<? super SingleEmitter<T>>)` | Yes | Yes | Yes | No |
+| `create(Consumer<? super FlowEmitter<T>>)` | Yes | Yes | No | No |
+| `create(Consumer<? super CompletionEmitter>)` | Yes | Yes | No | Yes |
+
+The first two `create` methods take an implementation of an interface which provides state and the generator methods:
+
+```java
+interface SyncGenerator<T, S> {
 
-Observable<T>.create(OnSubscribe<Observer<T>> onSubscribe)
+    S createState();
+    
+    S generate(S state, Observer<T> output);
+    
+    void disposeState(S state);
+}
 
-Single<T>.create(OnSubscribe<Single.Subscriber<T>> onSubscribe)
+interface AsyncGenerator<T, S> {
 
-Completable<T>.create(OnSubscribe<Completable.Subscriber<T>> onSubscribe)
+    S createState();
+    
+    S generate(S state, long requested, Observer<Observable<T>> output);
+    
+    void disposeState(S state);
+}
+```
+
+These latter three `create` methods will provide the following interaction interfaces to the `java.util.function.Consumer`:
+
+```java
+interface SingleEmitter<T> {
```

Comment:
Maybe I'm not understanding this correctly. Could you please give us an example of how this would work in practice?